docs/build/spells.md

```diff
@@ -0,0 +1,197 @@
+
+A spell consists of the following parts: 
+- The spell script, which will be executed
+- The list of trigger events (aka the trigger configuration) on which to run the spell
+- The initial state of the spell (e.g. initial state of spell's KV storage)
+
+After installation, the developer is able to update the initial settings: the spell script, the trigger configuration, and the state KV storage.
+
+Also, the developer is able to remove their spells, cleaning the state from the peer.
+
+Developer can check their spells' state: 
+- inspect their internal state (the KV storage, the script, the trigger config)
+- check if they’re currently subscribed to some triggers
+- obtain their peer identity: spell_id and worker_id.
+
+## Events that can trigger spells
+[Nox](https://github.com/fluencelabs/nox) triggers spells according to their configuration. Currently Nox provides the following event triggers:
+- Timer/CRON Triggers: run spells at a specified time or within a specified period.
+Users should be able to specify the following:
+    - when to run the spell,
+    - when to unsubscribe the spell from timer events,
+    - how often should the script be run.
+- Connection Triggers: run spells on connection/disconnection of some peers to the peer.
+
+All other trigger types can be implemented based on the timer triggers & the spell KV storage (blockchain event trigger, database event trigger, HTTP event trigger, etc)
+
+## Creating and updating spells
+
+Spells can be created and managed with [Fluence CLI](https://github.com/fluencelabs/cli). 
+
+Let's start from creating a new project with `fluence init` , which gives us a couple scaffolding choices:
+```bash
+fluence init
+? Select template (Use arrow keys)
+❯ quickstart
+  minimal
+  ts
+  js
+```
+
+Press return to select the default quickstart scaffolding option and enter *quickstart* as the project path when prompted:
+```bash
+? Enter project path or press enter to init in the current directory: quickstart
+Successfully initialized Fluence project template at /Users/arete/Documents/fluence/quickstart
+```
+
+Now let's create a new spell in a recently scaffolded Fluence CLI project by running `fluence spell new`:
+```bash
+quickstart $ fluence spell new
+? Enter spell name 1min_spell
+Successfully generated template for new spell at /Users/arete/Documents/fluence/quickstart/src/spells/1min_spell
+? Do you want to add spell 1min_spell to a default worker defaultWorker Yes
+Added 1min_spell to defaultWorker
+```
+
+Lets check what have been generated: a new spell directory was created under src/spells - 1min_spell:
+```bash
+.
+└── 1min_spell
+    ├── spell.aqua
+    └── spell.yaml
+```
+
+It contains [Aqua](/docs/aqua-book/introduction.md) script that will be executed and the spell.yaml file that specifies the trigger configuration for the spell.
```

Comment:
I'd include a link to the spell.yml schema https://github.com/fluencelabs/cli/blob/b8d6c5b9476962805a7fcf6bcc0cfb171089c584/docs/configs/spell.md